src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs

```diff
@@ -23,20 +28,43 @@ internal class VsCodeWindowManager : IVsCodeWindowManager, IVsCodeWindowEvents
             private readonly IVsCodeWindow _codeWindow;
             private readonly ComEventSink _sink;
             private readonly IOptionService _optionService;
+            private readonly IThreadingContext _threadingContext;
 
-            private INavigationBarController _navigationBarController;
-            private IVsDropdownBarClient _dropdownBarClient;
+            private INavigationBarController? _navigationBarController;
+            private IVsDropdownBarClient? _dropdownBarClient;
+            private WorkspaceRegistration _workspaceRegistration;
 
             public VsCodeWindowManager(TLanguageService languageService, IVsCodeWindow codeWindow)
             {
                 _languageService = languageService;
                 _codeWindow = codeWindow;
 
                 var workspace = languageService.Package.ComponentModel.GetService<VisualStudioWorkspace>();
```

Comment:
Now that we are fetching the workspace through the WorkspaceRegistration, can we just do _workspaceRegistration.Workspace.Services.GetService<IOptionService>() if/when we need it? My motivation here is mostly to make sure we aren't triggering a load of VisualStudioWorkspace sooner than necessary.

Reply:
I don't think I can do that easily because I need to hookup the option service OptionChanged event.  The workspace from workspace registration is null here and also can be null on AddAdornments, meaning we may not be able to get the option when adding adornments.  So I think I'd need to hook it up on workspace changed events to also add/remove dropdown based on the option.  

Since it's already loading the VS workspace this I'll leave it as-is in this PR unless you have a strong preference otherwise

Reply:
I guess my thought was when you get the workspace then you subscribe to OptionChanged. Technically™ this is wrong since if you have a non-VS workspace we're reading options from one workspace but subscribing to another.

Reply:
That makes sense.  I updated it to re-subscribe on add adornments and workspace changes.  And it will unsubscribe on remove adornments.